slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -221,18 +228,25 @@ ::grpc::Status SLAMServiceImpl::GetCurrentPointCloudMap(
     try {
         std::shared_lock optimization_lock{optimization_shared_mutex,
                                            std::defer_lock};
-        if (optimization_lock.try_lock()) {
+        if ((optimization_lock.try_lock()) && (!localization_map_ready)) {
```

Comment:
Same here, switch the order of checking localization_map_ready and locking, and remove extra parens.